src/librbd/mirror/snapshot/CreateNonPrimaryRequest.cc

```diff
@@ -120,10 +122,19 @@ void CreateNonPrimaryRequest<I>::handle_get_mirror_image(int r) {
     return;
   }
 
-  uuid_d uuid_gen;
-  uuid_gen.generate_random();
-  m_snap_name = ".mirror.non_primary." + mirror_image.global_image_id + "." +
-    uuid_gen.to_string();
+  std::stringstream snap_name_stream;
+  if (!m_group_snap_id.empty()) {
+    snap_name_stream << ".mirror.non_primary." << mirror_image.global_image_id
```

Comment:
Nit: Something like this would make it more obvious that the prefix is shared:
```
  std::stringstream ss;
  ss << ".mirror.non_primary." << mirror_image.global_image_id << ".";
  if (!m_group_snap_id.empty()) {
    ss << m_image_ctx->group_spec.pool_id << "_"
       << m_image_ctx->group_spec.group_id << "_"
       << m_group_snap_id;
  } else {
    uuid_d uuid_gen;
    uuid_gen.generate_random();
    ss << uuid_gen.to_string();
  }
```
I used a short variable name on purpose -- no need to be verbose given that there is no other stringstream in sight.